### PR TITLE
dev: bump node-exporter to 1.3.0

### PR DESCRIPTION
**Pull request by mayocream**
ref: #1225 

Patch changes to dev.
ref: #1235

### CODE REVIEW

**Comment by mayocream** [comment]
cc @oke-py

**Comment by inductor** [comment]
Since it's just DaemonSet I wouldn't merge this right away, but what are roughly changed since the current version?

File: manifests/infra/prometheus/overlays/development/kustomization.yaml [hunk 1]
```diff
@@ -5,3 +5,4 @@ resources:
 - ../../base
 patchesStrategicMerge:
 - service-selector.yaml
+- update-node-exporter.yaml
```
Comment (inductor) [inline manifests/infra/prometheus/overlays/development/kustomization.yaml:8]: I don't think this naming is good. What if you need additional update dependency?
Imo it has just be "node-exporter" or something more generic for future reference.

File: manifests/infra/prometheus/overlays/development/update-node-exporter.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      containers:
+        - args:
+            - --web.listen-address=127.0.0.1:9100
+            - --path.sysfs=/host/sys
+            - --path.rootfs=/host/root
+            - --no-collector.wifi
+            - --no-collector.hwmon
+            - --collector.filesystem.ignored-mount-points=^/(dev|proc|sys|var/lib/docker/.+|var/lib/kubelet/pods/.+)($|/)
+            - --collector.netclass.ignored-devices=^(veth.*|[a-f0-9]{15})$
+            - --collector.netdev.device-exclude=^(veth.*|[a-f0-9]{15})$
```
Comment (inductor) [inline manifests/infra/prometheus/overlays/development/update-node-exporter.yaml:23]: FYI the current indent is as follows 
Ref. https://github.com/cloudnativedaysjp/dreamkast-infra/blob/main/manifests/infra/prometheus/base/node-exporter.yaml#L78-L88

```suggestion
      containers:
      - args:
        - --web.listen-address=127.0.0.1:9100
        - --path.sysfs=/host/sys
        - --path.rootfs=/host/root
        - --no-collector.wifi
        - --no-collector.hwmon
        - --collector.filesystem.ignored-mount-points=^/(dev|proc|sys|var/lib/docker/.+|var/lib/kubelet/pods/.+)($|/)
        - --collector.netclass.ignored-devices=^(veth.*|[a-f0-9]{15})$
        - --collector.netdev.device-exclude=^(veth.*|[a-f0-9]{15})$
```

**Comment by mayocream** [comment]
Changes https://github.com/prometheus/node_exporter/releases/tag/v1.2.0:
> NOTE: Ignoring invalid network speed will be the default in 2.x
NOTE: Filesystem collector flags have been renamed. --collector.filesystem.ignored-mount-points is now --collector.filesystem.mount-points-exclude and --collector.filesystem.ignored-fs-types is now --collector.filesystem.fs-types-exclude. The old flags will be removed in 2.x.

**Comment by inductor** [comment]
@oke-py Looks like 1.3.0 is there too but do you have particular reasons you chose 1.2.2?

**Comment by inductor** [comment]
Regarding the version change, did you make sure that there's no change that you have to make for other resources for the following yaml files?

- https://github.com/cloudnativedaysjp/dreamkast-infra/blob/main/manifests/infra/prometheus/base/node-exporter.yaml
- https://github.com/cloudnativedaysjp/dreamkast-infra/blob/main/manifests/infra/prometheus/base/monitor.yaml

**Comment by mayocream** [comment]
@oke-py https://github.com/prometheus-operator/kube-prometheus/tree/v0.9.0/manifests still remain in the 1.2.2 version, the 1.3.0 version is in the main branch.

It seems to be only these changes that need to be patched for the new version. 
See #1235 for full changes diff between our files and kube-prometheus v0.9.0 version.

**Comment by mayocream** [comment]
@oke-py I compared kube-prometheus v0.9.0 version and its main branch, these no other changes since node-exporter 1.2.2.
These no breaking changes, so we can just update to the latest 1.3.0 version.

**Comment by mayocream** [comment]
> Regarding the version change, did you make sure that there's no change that you have to make for other resources in the following yaml files?
> 
> * https://github.com/cloudnativedaysjp/dreamkast-infra/blob/main/manifests/infra/prometheus/base/node-exporter.yaml
> * https://github.com/cloudnativedaysjp/dreamkast-infra/blob/main/manifests/infra/prometheus/base/monitor.yaml

The `app.kubernetes.io/version: 1.1.2` label should be replaced with `app.kubernetes.io/version: 1.3.0`, but I prefer to do it in the base manifest when updating the prod env. Any advice about it?

**Comment by inductor** [comment]
@oke-py Could you have a look and provide any advice?

**Review by oke-py — APPROVED** [review]
The mismatch in the label and the actual version would be confusing.
However, we can make a compromise.

**Review by inductor — APPROVED** [review]

**Pull request merged**